stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/MessageItem.kt

```diff
@@ -380,19 +383,47 @@ internal fun DefaultMessageItemCenterContent(
 
     val modifier = Modifier.widthIn(max = 250.dp)
 
-    MessageBubble(
-        modifier = modifier,
-        shape = messageBubbleShape,
-        color = messageBubbleColor,
-        content = {
-            MessageContent(
-                message = message,
-                onLongItemClick = onLongItemClick,
-                onGiphyActionClick = onGiphyActionClick,
-                onImagePreviewResult = onImagePreviewResult
+    val isFailed = ownsMessage && message.syncStatus == SyncStatus.FAILED_PERMANENTLY
+    if (!isFailed) {
+        MessageBubble(
+            modifier = modifier,
+            shape = messageBubbleShape,
+            color = messageBubbleColor,
+            content = {
+                MessageContent(
+                    message = message,
+                    onLongItemClick = onLongItemClick,
+                    onGiphyActionClick = onGiphyActionClick,
+                    onImagePreviewResult = onImagePreviewResult
+                )
+            }
+        )
+    } else {
+        Box(modifier = modifier) {
```

Comment:
I think we should put it in a box in both cases and onyl render the icon if the message has failed. We've had some issues previously where, because we had removed items from the box, Compose thinks things have been updated on a larger levels and items flicker.

Reply:
Yeah, this is a tradeoff I think. Let's consider failed messages very rare and consider that blinking is ok for that case. In return we get flat layout with slightly better performance.